Data/ByteString/Lazy.hs

```diff
@@ -1009,6 +1009,7 @@ findIndices k cs0 = findIndices' 0 cs0
   where findIndices' _ Empty        = []
         findIndices' n (Chunk c cs) = L.map ((+n).fromIntegral) (S.findIndices k c)
                              ++ findIndices' (n + fromIntegral (S.length c)) cs
+{-# INLINE findIndices #-}
```

Comment:
I wonder whether reimplementing the `Lazy` version via `S.findIndex`, returning elements one at a time from each initial non-empty chunk, and moving to the next chunk once the previous is exhausted, might not be more efficient by avoiding `++`, which might be costly as the number of chunks gets larger?

Reply:
Initial testing shows no measurable improvement from avoiding `++`, so I'm retracting this comment for now.  The tests did apply folds to the result, which perhaps optimise the `++` away, and one might invent some cases where this does not happen, but I am not going to pursue this...